osu.Game.Rulesets.Mania/Difficulty/Skills/Strain.cs

```diff
@@ -4,98 +4,281 @@
 #nullable disable
 
 using System;
-using osu.Framework.Utils;
 using osu.Game.Rulesets.Difficulty.Preprocessing;
 using osu.Game.Rulesets.Difficulty.Skills;
 using osu.Game.Rulesets.Mania.Difficulty.Preprocessing;
 using osu.Game.Rulesets.Mods;
 
+/* Rules of Strain Calculation
+ * 1) The current strain value can only depend on past notes
+ * 2) Notes on the same offset may see different strain values, as column order sorting is not guaranteed.
```

Comment:
I don't like referring to a note's time in the beatmap as "offset". Offset is relative to something. Notes aren't really offset from anything. The time value for each note is just where the note is, temporally, in the beatmap.

Reply:
Sure, i'll follow the repo's nomenclature